File: csharp/Agentic.Commerce.Protocol/Models/PaymentProviderType.cs
// this file is @generated
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace Agentic.Commerce.Protocol
{
    [JsonConverter(typeof(StringEnumConverter))]
    public enum PaymentProviderType
    {
        [EnumMember(Value = "stripe")]
        Stripe = 1,
    }
}

File: kotlin/src/main/kotlin/com/svix/agentic_commerce_protocol/PaymentProviderType.kt
// This file is @generated
package com.svix.agentic_commerce_protocol

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.encodeToJsonElement
import kotlinx.serialization.json.jsonPrimitive

@Serializable
enum class PaymentProviderType : ToQueryParam {
    @SerialName("stripe") STRIPE;

    override fun toQueryParam() = Json.encodeToJsonElement(this).jsonPrimitive.content
}
